tools/install-source/XamarinSourcesPathMangler.cs

```diff
@@ -0,0 +1,212 @@
+		static bool IsNativeType (string path)
+		{
+			return path.Contains (NativeTypeSubpath);
+		}
+
+		/// <summary>
+		/// Returns if the pat is from the runtime generated code.
+		/// </summary>
+		/// <returns><c>true</c>, if runrime was ised, <c>false</c> otherwise.</returns>
+		/// <param name="path">Path.</param>
+		static bool IsRunrime (string path)
+		{
+			return path.Contains (RuntimeSubpath);
+		}
+
+		/// <summary>
+		/// Returns the source path for a generated file.
+		/// </summary>
+		/// <returns>The source path for native type.</returns>
+		/// <param name="path">Path.</param>
+		string GetSourcePathForGeneratedPath (string path)
+		{
+			var frameworkPrefix = FrameworkPath.Remove (FrameworkPath.IndexOf (".framework", StringComparison.InvariantCulture));
+			var installPath = Path.Combine (InstallDir, "src", frameworkPrefix);
+			// we might be looking at a mdb that was already gone thorugh mdb rebase, if that is the case, do find the path for the final target 
+			if (path.StartsWith (Path.Combine (InstallDir, "src", frameworkPrefix), StringComparison.InvariantCulture)) {
+				var src = path.Substring (installPath.Length + 1);
+				src = Path.Combine (XamarinSourcePath, (InstallDir.Contains ("Xamarin.iOS.framework")) ? "/build/ios/native/" : "/build/mac/full/", src);
+				return src;
+			} else {
+				var nativeBuildPath = (InstallDir.Contains ("Xamarin.iOS.framework")) ? "/build/ios/native/" : "/build/mac/full/";
+				var pos = path.IndexOf (nativeBuildPath, StringComparison.InvariantCulture);
+				var src = path.Remove (0, pos + 1); // 3 for src and 1  for /
+				src = Path.Combine (XamarinSourcePath, src);
+				return src;
+			}
+		}
+
+		/// <summary>
+		/// Returns the source path for the native common types.
+		/// </summary>
+		/// <returns>The source path for native type.</returns>
+		/// <param name="path">Path.</param>
+		string GetSourcePathForNativeType (string path)
+		{
+			string src = "";
+			var pos = path.IndexOf (CommonSourceSubpath, StringComparison.InvariantCulture);
+			if (pos >= 0) {
+				src = path.Remove (0, pos + CommonSourceSubpath.Length);
+				src = Path.Combine (XamarinSourcePath, "build", "common", src);
+			} else {
+				pos = path.IndexOf (NativeTypeSubpath, StringComparison.InvariantCulture);
+				if (pos >= 0) {
+					src = path.Remove (0, pos);
+					src = Path.Combine (XamarinSourcePath, src);
+				} else {
+					Console.WriteLine ($"Ignoring path {path}");
+					return "";
+				}
+			}
+			return src;
+		}
+
+		/// <summary>
+		/// Returns the source path for the bindings that were manually done.
+		/// </summary>
+		/// <returns>The source path for manual source.</returns>
+		/// <param name="path">Path.</param>
+		string GetSourcePathForManualSource(string path)
+		{
+			var removalPath = $"/{FrameworkPath}/";
+			var srcFolder = Path.Combine ("src", FrameworkPath.Remove(FrameworkPath.IndexOf (".framework", StringComparison.InvariantCulture))) + Path.DirectorySeparatorChar;
```

Comment:
Style: space before opening parenthesis.

This can use `StringComparison.Ordinal`.